baggage/baggage.go

```diff
@@ -241,7 +242,12 @@ func NewMember(key, value string, props ...Property) (Member, error) {
 
 // NewMemberRaw returns a new Member from the passed arguments.
 //
-// The passed key must be compliant with W3C Baggage specification.
+// The passed key and value must be valid UTF-8 string.
```

Comment:
I think that `NewPropertyRaw` (and tests) should be updated as well.
The name/key passed to `NewPropertyRaw` can be any valid UTF-8 string.

Reply:
I am unsure about this, as the property is defined in the W3C baggage, not the otel specification. And, the W3C didn't allow UTF-8 in the key.

https://www.w3.org/TR/baggage/#key

> Additional metadata MAY be appended to values in the form of property set, represented as semi-colon ; delimited list of keys and/or key-value pairs, e.g. ;k1=v1;k2;k3=v3. Property keys and values are given no specific meaning by this specification. Leading and trailing OWS is allowed and is not considered to be a part of the property key or value.

Reply:
> property is defined in the W3C baggage, not the otel specification

You are right. The [OTel spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/baggage/api.md) does not even have the "Members" and "Property" distinction reflecting W3C definitions. I would even say that the OTel spec says "properties" to reflect "members" in W3C which makes it very confusing. 

I feel we should update the OTel spec first. Are you able to follow-up on this?

Reply:
I created https://github.com/open-telemetry/opentelemetry-specification/issues/3972

Reply:
I thought it was designed not to define "Property" from the spec, as it was only a W3C extension of baggage value. 

But yeah, the spec can mention the W3C property somewhere to prevent confusion. I can follow up on this.

Reply:
I missed this:

> `Metadata` Optional metadata associated with the name-value pair. This should be an opaque wrapper for a string with no semantic meaning. Left opaque to allow for future functionality.

So actually, I think it should have similar behavior and it is not actually blocked by specification.